### PR TITLE
Change type of pod environment config map to NamespacedName

### PR DESCRIPTION
**Pull request by FxKu**
Change type of `PodEnvironmentConfigMap` to `NamespacedName` to allow this config map in a namespace other than where the operator runs. If the ConfigMap isn't found in the configured default namespace than the operator looks up it's own namespace (if different) to have the same behavior as before. 

addresses #781

### CODE REVIEW

File: pkg/cluster/k8sres.go [hunk 1]
```diff
@@ -833,59 +834,65 @@ func (c *Cluster) generateStatefulSet(spec *acidv1.PostgresSpec) (*appsv1.Statef
...
-	if c.OpConfig.PodEnvironmentConfigMap != "" {
+	if c.OpConfig.PodEnvironmentConfigMap != (spec.NamespacedName{}) {
 		var cm *v1.ConfigMap
-		cm, err = c.KubeClient.ConfigMaps(c.Namespace).Get(c.OpConfig.PodEnvironmentConfigMap, metav1.GetOptions{})
+		cm, err = c.KubeClient.ConfigMaps(c.OpConfig.PodEnvironmentConfigMap.Namespace).Get(c.OpConfig.PodEnvironmentConfigMap.Name, metav1.GetOptions{})
 		if err != nil {
-			return nil, fmt.Errorf("could not read PodEnvironmentConfigMap: %v", err)
+			// if not found, try again using the operator namespace (old behavior)
+			if k8sutil.ResourceNotFound(err) && c.Namespace != c.OpConfig.PodEnvironmentConfigMap.Namespace {
+				cm, err = c.KubeClient.ConfigMaps(c.Namespace).Get(c.OpConfig.PodEnvironmentConfigMap.Name, metav1.GetOptions{})
```
Comment (sdudoladov) [inline pkg/cluster/k8sres.go]: so if the namespace of a configmap with additional env vars is is not specified, do we look in the operator namespace or the PG cluster namespace ? `c.Namespace` is the latter, but the comment states the former

**Comment by sdudoladov** [comment]
:+1:

**Comment by frittentheke** [comment]
@FxKu this is very cool thanks for changing this so quickly.

May I kindly ask if you plan on adding the propagation of those ENV variables from the configmap to all containers as well? (https://github.com/zalando/postgres-operator/issues/781)

**Comment by FxKu** [comment]
@frittentheke no concrete plans, yet. It's not on our current to do list. Shouldn't be too hard I guess. You still have the chance to open a PR :)

**Comment by sdudoladov** [comment]
:+1:

**Comment by FxKu** [comment]
:+1:

**Pull request merged**